### PR TITLE
New source api

### PR DESCRIPTION
**Pull request by reswqa**
see #164

### CODE REVIEW

**Review by jianyun8023 — COMMENTED** [review]
LGTM

File: pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            }
+        }
+        return pulsarAdmin;
+    }
+
+    @Nonnull
+    public PulsarClient getClient() {
+        if (pulsarClient == null) {
+            try {
+                pulsarClient = new ClientBuilderImpl(pulsarConfiguration).build();
```
[the remark this answers is not in the document]
Comment (reswqa) [reply pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java:174]: Can we replace it with the following code?
`pulsarClient = CachedPulsarClient.getOrCreate(pulsarConfiguration);`

File: pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            } catch (PulsarClientException e) {
+                throw new IllegalStateException("Cannot initialize pulsar admin", e);
```
[the remark this answers is not in the document]
Comment (reswqa) [reply pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java:176]: sure, it's a typo.

File: pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jianyun8023) [reply pulsar-flink-1.11/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSource.java:174]: Yes.

**Comment by reswqa** [comment]
Hi @jianyun8023, I've finished the modification. Do you think it's time to merge?

**Review by jianyun8023 — APPROVED** [review]

**Pull request merged**